atomix/cluster/src/main/java/io/atomix/raft/roles/LeaderRole.java

```diff
@@ -224,6 +227,29 @@ public CompletableFuture<ReconfigureResponse> onReconfigure(final ReconfigureReq
     return future;
   }
 
+  @Override
+  public CompletableFuture<ForceConfigureResponse> onForceConfigure(
+      final ForceConfigureRequest request) {
+    final Configuration currentConfiguration = raft.getCluster().getConfiguration();
+    if (currentConfiguration != null // this is not expected in a leader
+        && request.newMembers().containsAll(currentConfiguration.allMembers())
+        && currentConfiguration.allMembers().containsAll(request.newMembers())) {
+      // It is likely that the previous force configure was successfully completed, and this is a
+      // retry. So just respond success.
+      return CompletableFuture.completedFuture(
+          logResponse(
+              ForceConfigureResponse.builder()
+                  .withStatus(Status.OK)
+                  .withIndex(currentConfiguration.index())
+                  .withTerm(currentConfiguration.term())
+                  .build()));
+    }
+
+    // Do not force-configure when you are leader.
+    raft.transition(Role.FOLLOWER);
+    return super.onForceConfigure(request);
```

Comment:
:clown_face: Out of scope, but this reminds me I don't like the role hierarchy :sweat_smile: It's easy for this behavior to break in the future